### PR TITLE
Trim Async suffix on action names

### PR DESCRIPTION
**Pull request by pranavkm**
Fixes https://github.com/aspnet/AspNetCore/issues/4849

### CODE REVIEW

**Comment by davidfowl** [comment]
cc @BrennanConroy @halter73 We should do this in SignalR as well

**Review by dougbu — APPROVED** [review]

**Comment by rynowak** [comment]
What do you do if you want the action name to use async? `[ActionName(...)]`?

**Comment by dougbu** [comment]
> `[ActionName(...)]`?

Or, turn the new option off.

**Comment by pranavkm** [comment]
Hadn't thought about `ActionName`. It does make sense to leave it alone if the user explicitly specified it. However, since `ActionModel.ActionName` is constructed accounting for [`ActionNameAttribute`](https://github.com/aspnet/AspNetCore/blob/1808236dbc9a3272833af981c0cf874aba1bd952/src/Mvc/src/Microsoft.AspNetCore.Mvc.Core/ApplicationModels/DefaultApplicationModelProvider.cs#L304-L312) we wouldn't be able to support a convention for this. No biggie, but FYI since that would disappear in the next change

**Comment by pranavkm** [comment]
🆙 📅

File: src/Mvc/src/Microsoft.AspNetCore.Mvc.Core/MvcOptions.cs [hunk 1]
```diff
@@ -209,6 +209,21 @@ public int MaxModelValidationErrors
             }
         }
 
+        /// <summary>
+        /// Gets or sets a value that determines if MVC will ignore the suffix "Async" applied to
+        /// controller action names.
+        /// <para>
+        /// When <see langword="true"/>, MVC will strip action names of the Async suffix.
```
[the remark this answers is not in the document]
Comment (dougbu) [reply src/Mvc/src/Microsoft.AspNetCore.Mvc.Core/MvcOptions.cs]: Please fix this

File: src/Mvc/test/WebSites/BasicWebSite/Controllers/AsyncActionsController.cs [hunk 2]
```diff
@@ -26,6 +26,17 @@ public override void OnActionExecuted(ActionExecutedContext context)
             }
         }
 
+        public async Task<IActionResult> ActionWithSuffixAsync()
+        {
+            await Task.Yield();
+            return Ok();
+        }
+
+        public IActionResult ActionReturningViewAsync()
```
Comment (dougbu) [inline src/Mvc/test/WebSites/BasicWebSite/Controllers/AsyncActionsController.cs]: Very minor but it looks strange to have a sync method named with an Async suffix.

**Review by dougbu — APPROVED** [review]
Still looks good to me

**Pull request merged**